### ISSUE TITLE
p-listbox with Checkboxes Accessibility Issues

### ISSUE THREAD

**Issue by aciesla4** [primefaces/primeng#18607]
### Describe the bug

The p-listbox component with checkboxes has accessibility violations when running the IBM Equal Accessibility Checker. The two issues are: Each form control must have an associated label and UI component must have at least one tabbable descendant for keyboard access. 

### Reason for not contributing a PR

- [x] Lack of time
- [ ] Unsure how to implement the fix/feature
- [ ] Difficulty understanding the codebase
- [ ] Other

### Reproducer

https://gwdkjoae-4bbo9nyp.stackblitz.io

### Environment

Stackblitz

### Angular version

20.0.4

### PrimeNG version

v20

### Steps to reproduce the behavior

1. Install the [IBM Equal Accessibility Checker](https://chromewebstore.google.com/detail/ibm-equal-access-accessib/lkcagbfjnkomcinoddgooolagloogehp)
2. Open the stackblitz
3. Run the accessibility checker
4. Filter on violations only and note the errors: 'Form control element <input> has no associated label' and 'None of the descendent elements with "option" role is tabbable'

### Expected behavior

There are no accessibility violations for the p-listbox component when checkboxes are used

**Comment by cagataycivici** [comment]
Your PR is appreciated.
